pkg/generator/ingress_translator.go

```diff
@@ -106,6 +113,17 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 	externalTLSHosts := make([]*route.VirtualHost, 0, len(ingress.Spec.Rules))
 	clusters := make([]*v3.Cluster, 0, len(ingress.Spec.Rules))
 
+	cfg := config.FromContext(ctx)
```

Comment:
No, I added [this](https://github.com/knative-extensions/net-kourier/pull/1171/files#diff-1c1c1dc4a9824c0156196bdf2594574cefc8099a0a67e2e392044468bc6f6a5dR341) to the controller, so we have the CMs now properly on the context, before calling the ingress translator.

Reply:
Is it possible that the cm will not be deployed yet when we reach the kubeclient call. I was reading again my comment [here](https://github.com/knative-extensions/net-kourier/pull/959#discussion_r1048441513), so sharedMain has a cm watcher that will wait for cms to exist and even if we try to [fetch certain cms earlier and fail (since they dont exist)](https://github.com/knative/pkg/blob/main/injection/sharedmain/main.go#L408-L410) we will get them finally due to the cmw.Start() call.

Reply:
I guess the idea was to be eventually be consistent. However, I am in favor of fixing it properly. I would try the wait approach so we dont have to think about corner cases.